docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,200 @@
+	    // such as getting the location from the device.
+	    var getDirectionsTask = _directionsClient.GetDrivingDirectionsToLocation(user.LastKnownLocation);
+		
+	    // This independent work can be done concurrently
+	    // since it doesn't rely on directions!
+	    NotifyUserOfPickupAsync(user.Contact);
+	    
+	    // The await operator suspends SelectUserForPickup_Pressed, returning control to its caller.
+	    // This is what allows the app to be responsive and not hang on the UI thread.
+	    var directionsJson = await getDirectionsTask;
+		
+	    // Using the JSON.NET library to deserialize data
+	    var directionsData = JsonConvert.DeserializeObject<DirectionsData>(directionsJson);
+	    		    
+	    DrawRouteOnMap(directionsData.Polyline);
+	}
+
+Web service snippet:
+
+.. code-block:: c#
+
+	[HttpGet]
+	public async Task<string> GetDirections(GeoPoint startPos, GeoPoint endPos)
+	{
+	    // Suspends GetUser() to allow the caller (the web service) to accept another request,
+	    // rather than blocking on this one.
+	    var directions = await _directionsService.GetDirectionsAsync(startPos, endPos);
+	    
+	    // Using the JSON.NET library to serialize data
+	    var json = JsonConvert.SerializeObject(directions);
+	    
+	    return json;
+	}
+	
+Bigger Example (VB)
+-------------------------
+
+These are the VB-equivalent code snippets from above.
+
+Client app snippet:
+
+.. code-block:: vb.net
+
+	Private Async Sub GetDirectionsForPickup_Pressed(sender As Object, e As CustomUserDataEventArgs) Handles GetDirectionsForPickup.Click
+		
+		Dim user As BasicUserData = e.Source
+		
+		' _directionsClient is a made-up object that would encapsulate certain complexity,
+		' such as getting the location from the device.
+		Dim getDirectionsTask As Task(Of String) = _directionsClient.GetDrivingDirectionsToLocation(user.LastKnownLocation)
+		
+		' This independent work can be done concurrently
+		' since it doesn't rely on directions!
+		NotifyUserOfPickupAsync(user.ContactInfo)
+		
+		Dim directionsJson As String = Await getDirectionsTask
+		
+		' Using the JSON.NET library to deserialize data
+		Dim d As DirectionsData = JsonConvert.DeserializeObject(Of DirectionsData)(directionsJson)
+		
+		DrawRouteOnMap(d.Polyline)		
+	End Sub
+
+Web Service snippet:
+
+.. code-block:: vb.net
+
+	<HttpGet>
+	Public Async Function GetDirections(startPos as GeoPoint, endPos as GeoPoint) As Task(Of String)
+
+		' Suspends GetUser() to allow the caller (the web service) to accept another request,
+		' rather than blocking on this one.
+		Dim dirs As Directions = Await _directionsService.GetDirectionsAsync(startPos, endPost);
+		
+		Dim json As String = JsonConvert.SerializeObject(dirs);
+		
+		Return json
+	End Function
+	
+More on Task and Task<T>
+------------------------
+
+As mentioned before, Tasks are constructs which represent long-running operations calling code must wait on.
```

Comment:
There is no specific affinity between Tasks and "long-running operations". LROs is one of the use cases.